components/director/internal/statusupdate/handler.go

```diff
@@ -0,0 +1,120 @@
+package statusupdate
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/pkg/errors"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/persistence"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/kyma-incubator/compass/components/director/internal/consumer"
+)
+
+type update struct {
+	table    Table
+	transact persistence.Transactioner
+	repo     StatusUpdateRepository
+}
+
+type Table string
+
+//go:generate mockery -name=StatusUpdateRepository -output=automock -outpkg=automock -case=underscore
+type StatusUpdateRepository interface {
+	UpdateStatus(ctx context.Context, id, table string) error
+	IsConnected(ctx context.Context, id, table string) (bool, error)
+}
+
+const (
+	applicationsTable Table = "applications"
+	runtimesTable     Table = "runtimes"
+)
+
+func New(transact persistence.Transactioner, repo StatusUpdateRepository) *update {
+	return &update{
+		table:    "",
+		transact: transact,
+		repo:     repo,
+	}
+}
+
+func (u *update) Handler() func(next http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			ctx := r.Context()
+			consumerInfo, err := consumer.LoadFromContext(ctx)
+			if err != nil {
+				u.writeError(w, errors.Wrap(err, "while fetching consumer info from from context").Error(), http.StatusBadRequest)
+				return
+			}
+
+			switch consumerInfo.ConsumerType {
+			case consumer.Application:
+				u.table = applicationsTable
+			case consumer.Runtime:
+				u.table = runtimesTable
+			default:
+				next.ServeHTTP(w, r.WithContext(ctx))
+				return
+			}
+
+			tx, err := u.transact.Begin()
+			if err != nil {
+				u.writeError(w, errors.Wrap(err, "while opening transaction").Error(), http.StatusInternalServerError)
+				return
```

Comment:
updating status is only auxiliary functionality. Because of that, even if we got an error I suggest continuing handling request. IMO, we should always call `next.ServeHTTP`, do not write errors in the response (just log them). @crabtree @pkosiec what do you think?